packages/trader/src/Modules/Contract/Components/LastDigitPrediction/last-digit-prediction.tsx

```diff
@@ -79,11 +79,11 @@ const LastDigitPrediction = ({
             [key: string]: (val: number | null) => boolean;
         } = {
             DIGITMATCH: (val: number | null) => val === barrier,
-            DIGITDIFF: (val: number | null) => val !== barrier && !!barrier && !isNaN(barrier),
-            DIGITOVER: (val: number | null) => !!(val && barrier) && val > barrier,
-            DIGITUNDER: (val: number | null) => !!(val && barrier) && val < barrier,
+            DIGITDIFF: (val: number | null) => val !== barrier,
+            DIGITOVER: (val: number | null) => !!((val || val === 0) && (barrier || barrier === 0)) && val > barrier,
+            DIGITUNDER: (val: number | null) => !!((val || val === 0) && (barrier || barrier === 0)) && val < barrier,
             DIGITODD: (val: number | null) => !!val && Boolean(val % 2),
-            DIGITEVEN: (val: number | null) => !!val && !(val % 2),
+            DIGITEVEN: (val: number | null) => (!!val && !(val % 2)) || val === 0,
```

Comment:
`typeof val === 'number' && !(val % 2);`
Not sure if it's better, please, ignore if not relevant

Reply:
Same here, there's no difference between the solutions. It's just a preference

Reply:
Yep, but I was thinking about readability, because  `(!!val && !(val % 2)) || val === 0` is a bit hard for understanding))

Reply:
Honestly, I agree about `readability`, but for me it is more readable without having the type check inside. That feels easier for me to understand it by looking at operands and operators knowing that val is already a number. I don't know maybe it's just me..